### PR TITLE
Add | Adding net8 support

### PR DESCRIPTION
**Pull request by JRahnama**
Replacing Net7 with Net8.0

### CODE REVIEW

**Comment by H-Yeo** [comment]
In CertificateTest.cs, do we want to clean up this comment as well?
// Net7.0 the new API added by dotnet runtime will check SANS and then SubjectName
to maybe
//  >=Net7.0 the new API added by dotnet runtime will check SANS and then SubjectName

**Comment by H-Yeo** [comment]
Everything else looks good to me.

**Comment by JRahnama** [comment]
> In CertificateTest.cs, do we want to clean up this comment as well? // Net7.0 the new API added by dotnet runtime will check SANS and then SubjectName to maybe // >=Net7.0 the new API added by dotnet runtime will check SANS and then SubjectName

Basically the API was added in net7, so it could stay the same or I can adjust it to net7 and greater version.

File: tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec
Comment (campersau) [inline tools/specs/add-ons/Microsoft.Data.SqlClient.AlwaysEncrypted.AzureKeyVaultProvider.nuspec:37]: `</group>` missing after resolving merge conflict.

File: src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs:1452]: Based on [.net doc](https://learn.microsoft.com/en-us/dotnet/api/system.exception.getobjectdata?view=net-8.0&devlangs=csharp&f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(System.Exception.GetObjectData)%3Bk(SolutionItemsProject)%3Bk(DevLang-csharp)%26rd%3Dtrue) suggestion.

Should this be applied on API docs as well?
cc @David-Engel

```suggestion
        [Obsolete("This API supports obsolete formatter-based serialization. It should not be called or extended by application code.", DiagnosticId="SYSLIB0051", UrlFormat="https://aka.ms/dotnet-warnings/{0}")]
```

File: src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlException.cs
Comment (DavoudEshtehari) [inline src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlException.cs:66]:
```suggestion
        [Obsolete("This API supports obsolete formatter-based serialization. It should not be called or extended by application code.", DiagnosticId="SYSLIB0051", UrlFormat="https://aka.ms/dotnet-warnings/{0}")]
```

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/Batch/BatchTests.cs
Comment (JRahnama) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/Batch/BatchTests.cs:59]: I just addressed the failures on net8 and did not check the rest. 
Basically in my opinion this fix should have had a separate PR to address the test fix, but since it just showed up with net8 changes I included the fix in this PR.

Comment (David-Engel) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/Batch/BatchTests.cs:59]: > Just a comment here: In general, it appears this class uses curly braces { ... } for large code blocks, with the exception of nested usings. I'm indifferent on whether you change this or not.

@H-Yeo 
There is a lot of existing code with that style. When C# 8 eliminated the braces requirement, we started allowing it in code and using it more heavily in test code. There's no need to keep new code consistent with old code in this respect and contributors are free to convert old style to new as they see fit. In non-test code, we are more thoughtful about the using scope so we tend to keep the braces for clarity, but they aren't strictly required.

**Review by DavoudEshtehari — COMMENTED** [review]
LGTM, waiting for the feedback from the API doc process.

**Comment by robertmclaws** [comment]
Hopefully with the new downstream vulnerabilities that have been issues, this package will get updated soon.

Moving forward, this package should contain multiple builds targeting the currently supported runtimes, so that people installing this package in their apps can get the latest release for the .NET runtime they are targeting. I was SHOCKED to see one of my NuGet packages was resolving references all the way back down to .NET 6, even though I was targeting .NET 8, because of this package.

Thanks for your swift attention to getting this resolved!

**Review by DavoudEshtehari — APPROVED** [review]
🎉

**Pull request merged**
